### PR TITLE
feat(batch): add spill at least memory for hash agg

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Related issue https://github.com/risingwavelabs/risingwave/issues/16615
- Add spill at least memory for HashAgg, so the algorithm won't recursively spill.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/batch/src/executor/hash_agg.rs [hunk 1]
```diff
@@ -470,11 +491,18 @@ impl AggSpillManager {
...
 impl<K: HashKey + Send + Sync> HashAggExecutor<K> {
     #[try_stream(boxed, ok = DataChunk, error = BatchError)]
     async fn do_execute(self: Box<Self>) {
         let child_schema = self.child.schema().clone();
         let mut need_to_spill = false;
+        // If the memory upper bound is less than 1MB, we don't need to check memory usage.
+        let check_memory = match self.memory_upper_bound {
+            Some(upper_bound) => upper_bound > SPILL_AT_LEAST_MEMORY,
+            None => true,
+        };
```
Comment (fuyufjh) [inline src/batch/src/executor/hash_agg.rs:505]: The result seems to remain consistent during the lifetime of `HashAggExecutor`, right? If so, why not just use `bool check_memory` as a struct field?

Comment (chenzl25) [reply src/batch/src/executor/hash_agg.rs:505]: Because this information could be used to decide a partition number in the future.

Comment (fuyufjh) [reply src/batch/src/executor/hash_agg.rs:505]: I see

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**